src/Renderer/UI/SimulationView.fs

```diff
@@ -378,23 +384,148 @@ let private viewStatefulComponents step comps numBase model dispatch =
+                                model.Sheet.ChangeCounterComp sheetDispatch (ComponentId comp.Id) (CounterNoEnable w)
+                            | _ -> ()
+                    | _ -> ())
+                endSimulation endDispatch dispatch
             div [] [
-                str <| "Error found in sheet '" + dep + "' which is a dependency:"
+                str simError.Msg
                 br []
+                ul [] reacListOfCompsAffected
+                Button.button [
+                    Button.Color IsSuccess
+                    Button.OnClick (fun _ -> removeInPorts())
+                ] [str "Remove removable input ports from component properties"]
+            ]
+        | UnnecessaryNC ->
+            let removeNCAndChangeAdderType() =
+                // delete NotConnected components
+                symbolDispatch <| SymbolT.DeleteSymbols (simError.ConnectionsAffected
+                    |> List.collect (fun (ConnectionId connId) ->
+                        match tryGetConnectionById connId with
+                        | Some conn -> [ComponentId conn.Target.HostId]
+                        | None -> []))
+                // delete affected connections
+                busWireDispatch <| BusWireT.DeleteWires simError.ConnectionsAffected
+
+                simError.ComponentsAffected
+                |> List.collect (fun (ComponentId compId) -> Option.toList (tryGetComponentById compId))
+                |> List.iter (fun comp ->
+                    match comp.Type with
+                    | NbitsAdder w -> model.Sheet.ChangeAdderComp sheetDispatch (ComponentId comp.Id) (NbitsAdderNoCout w)
+                    | NbitsAdderNoCin w -> model.Sheet.ChangeAdderComp sheetDispatch (ComponentId comp.Id) (NbitsAdderNoCinCout w)
+                    | _ -> failwithf "Unexpected adder type. Should only encounter these 2 types with this error message")
+                endSimulation endDispatch dispatch
+
+            div [] [
                 str simError.Msg
                 br []
-                str <| "Please fix the error in this sheet and retry."
+                ul [] reacListOfCompsAffected
+                Button.button [
+                    Button.Color IsSuccess
+                    Button.OnClick (fun _ -> removeNCAndChangeAdderType())
+                ] [str "Remove unnecessary 'Not Connected' components"]
             ]
+        | _ ->
+            match simError.InDependency with
+            | None ->
+                div [] [
+                    str simError.Msg
+                    br []
+                    str <| "Please fix the error and retry."
+                ]
+            | Some dep ->
+                div [] [
+                    str <| "Error found in sheet '" + dep + "' which is a dependency:"
+                    br []
+                    str simError.Msg
+                    br []
+                    str <| "Please fix the error in this sheet and retry."
+                ]
```

Comment:
Overall - addNCremoveports does a lot of work.
It should eb top-level
It should be made more concise and readable by factoring out as functions things which are repeated. The irregularity exists - but it can be isolated to a match statement in a function somewhere. Doing this well is non-trivial so we maybe need a teams meeting after you have thought about my comments. Also you could look at the IDC document on guide to code quality under IDC resources (I realise it needs example).

Benefits for doing this well:
(1) improves your programming
(2) makes Issie codebase cleaner